run_md.py

```diff
@@ -0,0 +1,879 @@
+        out.write(f'{molid}\t{resid}\n')
+    with open(os.path.join(wdir_ligand, 'all_resid.txt'), 'a') as out:
+        out.write(f'{molid}\t{resid}\n')
+
+    return wdir_ligand_cur, molid, resid
+
+def prep_md_files(var_lig_data, protein_name, system_lig_data, wdir_protein, wdir_md, clean_previous=False):
+    def copy_md_files_to_wdir(files, wdir_copy_to):
+        for file in files:
+            shutil.copy(file, os.path.join(wdir_copy_to, os.path.basename(file)))
+
+    if var_lig_data:
+        wdir_var_ligand_cur, var_lig_molid, var_lig_resid = var_lig_data
+        wdir_md_cur = os.path.join(wdir_md, f'{protein_name}_{var_lig_molid}')
+    else:
+        wdir_md_cur = os.path.join(wdir_md, protein_name)
+
+    if clean_previous and os.path.isdir(wdir_md_cur):
+        shutil.rmtree(wdir_md_cur)
+
+    os.makedirs(wdir_md_cur, exist_ok=True)
+
+    # topol for protein for all chains
+    copy_md_files_to_wdir(glob(os.path.join(wdir_protein,'*.itp')), wdir_copy_to=wdir_md_cur)
+    copy_md_files_to_wdir([os.path.join(wdir_protein, "topol.top")], wdir_copy_to=wdir_md_cur)
+
+    # prep ligands and cofactor
+    # copy ligand.itp to md_wdir_cur. Will be edited
+    if var_lig_data:
+        wdir_var_ligand_cur, var_lig_molid, var_lig_resid = var_lig_data
+        all_itp_list, all_gro_list, all_posres_list, all_lig_molids, all_resids = [os.path.join(wdir_md_cur, f'{var_lig_molid}.itp')],\
+                                                     [os.path.join(wdir_var_ligand_cur, f'{var_lig_molid}.gro')],\
+                                                     [os.path.join(wdir_var_ligand_cur, f'posre_{var_lig_molid}.itp')],\
+                                                     [var_lig_molid], \
+                                                     [var_lig_resid]
+        copy_md_files_to_wdir([os.path.join(wdir_var_ligand_cur, f'{var_lig_molid}.itp')], wdir_copy_to=wdir_md_cur)
+    else:
+        all_itp_list, all_gro_list, all_posres_list, all_lig_molids, all_resids = [], [], [], [], []
+
+    # copy system_lig itp to ligand_md_wdir
+    for wdir_system_ligand_cur, system_lig_molid, system_lig_resid in system_lig_data:
+        copy_md_files_to_wdir([os.path.join(wdir_system_ligand_cur, f'{system_lig_molid}.itp')], wdir_copy_to=wdir_md_cur)
+        all_itp_list.append(os.path.join(wdir_md_cur, f'{system_lig_molid}.itp'))
+        all_gro_list.append(os.path.join(wdir_system_ligand_cur, f'{system_lig_molid}.gro'))
+        all_posres_list.append(os.path.join(wdir_system_ligand_cur, f'posre_{system_lig_molid}.itp'))
+        all_lig_molids.append(system_lig_molid)
+        all_resids.append(system_lig_resid)
+
+    return wdir_md_cur, all_itp_list, all_gro_list, all_posres_list, all_lig_molids, all_resids
+
+
+def supply_mols(fname, set_resid=None):
+    def create_random_resid():
+        # gro
+        ascii_uppercase_digits = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+        return ''.join(random.choices(ascii_uppercase_digits, k=3))
```

Comment:
This is a wrong approach. Names should not be random, otherwise you have collisions, that is completely forbidden. Create a function (generator) which will return next name as a combination of symbols
```python
for i in itertools.product(ascii_uppercase_digits, repeat = 3):
    yield ''.join(i)
```

other code should be changed accordingly

Reply:
your version is working, but unnecessary complex

Reply:
The next step I check if such combination was already used. Anyway I will fix it